tests/yast2_cmd/yast_storage.pm

```diff
@@ -0,0 +1,77 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: yast storage test
+# List disks and list partitions
+# Maintainer: Michael Grifalconi <mgrifalconi@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+    zypper_call "in yast2-storage";
+    #    assert_script_run 'yast disk list disks';
+    #    assert_script_run 'yast disk list partitions';
+    #
+    my $lsblk_output      = script_output("lsblk");
+    my $list_disks_output = script_output("yast disk list disks");
+    my $list_part_output  = script_output("yast disk list partitions");
+
+    # Checks that all disks listed by 'lsblk' are part of 'yast disk list disks'
+    my $current_test_device;
+    my $found;
+    my @tested_devices;
+    for my $lsblk_line (split /^/, $lsblk_output) {
+        if ($lsblk_line =~ /^([a-z]d[a-z])(.*)disk/) {
+            $current_test_device = $1;
+            $found               = 0;
+            for my $list_disks_line (split /^/, $list_disks_output) {
+                if ($list_disks_line =~ /^(.*)$current_test_device/) {
+                    $found++;
+                    push @tested_devices, $current_test_device;
+                }
+            }
+            die "Could not find disk match between lsblk and yast disk list disks" unless ($found);
+        }
+    }
+    die "Could not find a valid block device to test" unless (@tested_devices);
+
+    # Checks that all partitions listed by 'lsblk' are part of 'yast disk list partitions'
+    my @part_not_matched;
+    @tested_devices = ();
+
+    for my $lsblk_line (split /^/, $lsblk_output) {
+        if ($lsblk_line =~ /([a-z]d[a-z][1-9]+)(.*)part/) {
+            $current_test_device = $1;
+            $found               = 0;
+            for my $list_part_line (split /^/, $list_part_output) {
+                if ($list_part_line =~ /^(.*)$current_test_device/) {
+                    $found++;
+                    push @tested_devices, $current_test_device;
+                }
+            }
+            unless ($found) {
+                push @part_not_matched, $current_test_device;
+            }
+        }
+    }
+    die "Could not find a valid partition to test" unless (@tested_devices);
+
+    # Some partitions may not be listed in 'yast disk list partitions' due to a known issue.
+    if (@part_not_matched) {
+        record_soft_failure 'bsc#1150450';
+    }
```

Comment:
I don't think this is correct as [bsc#1150450](https://bugzilla.suse.com/show_bug.cgi?id=1150450) was marked as invalid because "This old yast-storage client only shows the partitions with a known filesystem." so I suggest to check only for filesystems we're using, which means swap, btrfs, xfs and maybe also ext2,3,4. Is this possible?

Reply:
Like we discussed in the chat:
It is possible to enhance it further, but it's probably not worth investing since there is no 'strage' filesystem in the openQA vm which runs the test, therefore this case will most likely never happen.
If the situation changes and we start to see softfails, we can change it then.